storage/range.go

```diff
@@ -940,6 +939,8 @@ func (r *Range) EndTransaction(batch engine.Engine, args *proto.EndTransactionRe
 			reply.SetGoError(r.splitTrigger(batch, args.SplitTrigger))
 		} else if args.MergeTrigger != nil {
 			reply.SetGoError(r.mergeTrigger(batch, args.MergeTrigger))
+		} else if args.ChangeReplicasTrigger != nil {
+			// ChangeReplicasTrigger is processed on input, not output.
```

Comment:
Maybe factor out the code in ChangeReplicas which creates the updated desc and call it both from ChangeReplicas and from changeReplicasTrigger.